.changeset/chatty-beds-doubt.md

```diff
@@ -0,0 +1,41 @@
+---
+'@clerk/clerk-js': minor
+'@clerk/backend': minor
+'@clerk/nextjs': minor
+'@clerk/shared': minor
+'@clerk/clerk-react': minor
+'@clerk/types': minor
+---
+
+Support for permission checks with Gate.
+
+Example usage:
+//index.d.ts
+```ts
+declare global {
+    interface OrganizationCustomPermissions {
+      "org:appointment:accept":"org:appointment:accept";
+      "org:appointment:decline":"org:appointment:decline";
+      "org:patients:create":"org:patients:create";
+    }
+
+    interface OrganizationCustomPermissions {
+      "org:doctor":"org:doctor";
+      "org:nurse":"org:nurse";
+    }
```

Comment:
❓ What's the thought behind making this an interface that has identical keys / values, vs. something like `"org:appointment:accept" | "org:appointment:decline" | "org:appointment:create"`?

Reply:
This was the end result i wanted to have, unfortunately didn't manage to make it work.

I was getting `duplicate identifier` errors, as types seem they cannot be overridden.